widgetastic_manageiq.py

```diff
@@ -425,3 +431,145 @@ class Accordion(PFAccordion):
     def is_dimmed(self):
         return bool(
             self.browser.elements('.//div[contains(@id, "tree") and contains(@class, "dimmed")]'))
+
+
+class Paginator(Widget):
+    """
+    Represents Paginator control that includes First/Last/Next/Prev buttons and control displaying
+    amount of items on current page and overall amount.
+
+    It is mainly used in Paginator Pane.
+    """
+    PAGINATOR_LOCATOR = './/ul[@class="pagination"]/'
+    CUR_PAGE_LOCATOR = PAGINATOR_LOCATOR + './li/span/input[@name="limitstart"]/..'
+    PAGE_BUTTON_LOCATOR = PAGINATOR_LOCATOR + './li[contains(@class, {})]/span'
+
+    def __init__(self, parent, logger=None):
+        super(Paginator, self).__init__(parent=parent, logger=logger)
+
+    def __locator__(self):
+        return self.PAGINATOR_LOCATOR
+
+    def _is_enabled(self, locator):
+        el = self.browser.element(locator + '/..')
+        return 'disabled' not in self.browser.classes(el)
+
+    def _click_button(self, locator):
+        if self._is_enabled(locator):
+            self.browser.click(locator)
+        else:
+            raise NoSuchElementException('such button is either absent or grayed out')
+
+    def next(self):
+        self._click_button(self.PAGE_BUTTON_LOCATOR.format(quote('next')))
+
+    def prev(self):
+        self._click_button(self.PAGE_BUTTON_LOCATOR.format(quote('prev')))
+
+    def last(self):
+        self._click_button(self.PAGE_BUTTON_LOCATOR.format(quote('last')))
+
+    def first(self):
+        self._click_button(self.PAGE_BUTTON_LOCATOR.format(quote('first')))
```

Comment:
If you pass the PAGE_BUTTON_LOCATOR 4 times into the same function and never anything else, I would rather think of passing just the string and let the `_click_button` do it.

By the way, `next` is used in iterators, so I would personally name the methods like `click_next` and such.